src/node.cc

```diff
@@ -234,124 +237,156 @@ MaybeLocal<Value> ExecuteBootstrapper(Environment* env,
   return scope.EscapeMaybe(result);
 }
 
-MaybeLocal<Value> RunBootstrapping(Environment* env) {
-  CHECK(!env->has_run_bootstrapping_code());
+#if HAVE_INSPECTOR && NODE_USE_V8_PLATFORM
+int Environment::InitializeInspector(
+    inspector::ParentInspectorHandle* parent_handle) {
+  std::string inspector_path;
+  if (parent_handle != nullptr) {
+    DCHECK(!is_main_thread());
+    inspector_path = parent_handle->url();
+    inspector_agent_->SetParentHandle(
+        std::unique_ptr<inspector::ParentInspectorHandle>(parent_handle));
+  } else {
+    inspector_path = argv_.size() > 1 ? argv_[1].c_str() : "";
+  }
 
-  EscapableHandleScope scope(env->isolate());
-  Isolate* isolate = env->isolate();
-  Local<Context> context = env->context();
+  isolate_->GetHeapProfiler()->AddBuildEmbedderGraphCallback(
+      Environment::BuildEmbedderGraph, this);
```

Comment:
This is dependent on `#if HAVE_INSPECTOR` now, but you can do heap dumps without the inspector, so should we maybe not do this here? (Sorry that I’m bringing it up again :smile:)

Reply:
An right, maybe it also makes sense to move it back to `InitializeDiagnostics` and then we invoke `InitializeDiagnostics`  before `InitializeInspector`. The other part of `InitializeDiagnostics` is adding GC callbacks for dtrace, I think it also make sense to do that before starting the inspector agent.